moobot/events.py

```diff
@@ -140,16 +141,38 @@ async def send_event_announcement(client: discord.Client, event: MoobloomEvent)
     announcement_channel = get_announcement_channel(client)
 
     message = await announcement_channel.send(embed=build_event_announcement_embed(event))
-    await message.add_reaction(settings.rsvp_yes_emoji)
-    await message.add_reaction(settings.rsvp_maybe_emoji)
-    await message.add_reaction(settings.rsvp_no_emoji)
 
     with Session() as session:
         session.add(event)
         event.announcement_message_id = str(message.id)
         session.commit()
 
 
+async def add_event_rsvp_emojis(client: discord.Client) -> None:
+    with Session() as session:
+        events: list[
+            MoobloomEvent
+        ] = (  # if we really care about the extra api calls we can add a state filter here
+            session.query(MoobloomEvent).filter(MoobloomEvent.deleted == False).all()
+        )
+
+        await asyncio.gather(*[populate_event_emojis(client, event) for event in events])
+
+        session.commit()
+
+
+async def populate_event_emojis(
+    client: discord.Client, event: MoobloomEvent
+) -> None:  # taking name suggestions
+    announcement_channel = get_announcement_channel(client)
+    message = await announcement_channel.fetch_message(event.announcement_message_id)
+    if len(message.reactions) == 0:
```

Comment:
what if a user reacts to the event before the bot gets here?

Reply:
maybe filter for reacts by the bot

Reply:
how? users don't have the reaction permission on #event-announce; the bot is the only account with permissions to add a new reaction

Reply:
a more robust way to do this would indeed be to check if the bot has placed each kind of reaction, if you don't want to rely on the permissions

Reply:
oh interesting, i forgot i did that. still I think it's worth adding the check here just because it's free and i don't want to assume a specific channel config like that will exist

Reply:
plus like someone with permissions like me can still react too fast and break it

Reply:
> a more robust way to do this would indeed be to check if the bot has placed each kind of reaction, if you don't want to rely on the permissions

I think that would be fine, it doesn't require any additional api calls does it?

Reply:
we don't have to do that though

Reply:
these checks are made redundant by the state tracking in the DB in the latest commit